src/pulse_cnt.rs

```diff
@@ -0,0 +1,533 @@
+        self.0.invert_edge_input()
+    }
+
+    #[inline]
+    pub fn set_invert_edge_input(&mut self, val: u32) {
+        self.0.set_invert_edge_input(val)
+    }
+
+    #[inline]
+    pub fn invert_level_input(&self) -> u32 {
+        self.0.invert_level_input()
+    }
+
+    #[inline]
+    pub fn set_invert_level_input(&mut self, val: u32) {
+        self.0.set_invert_level_input(val)
+    }
+
+    #[inline]
+    pub fn virt_edge_io_level(&self) -> u32 {
+        self.0.virt_edge_io_level()
+    }
+
+    #[inline]
+    pub fn set_virt_edge_io_level(&mut self, val: u32) {
+        self.0.set_virt_edge_io_level(val)
+    }
+
+    #[inline]
+    pub fn virt_level_io_level(&self) -> u32 {
+        self.0.virt_level_io_level()
+    }
+
+    #[inline]
+    pub fn set_virt_level_io_level(&mut self, val: u32) {
+        self.0.set_virt_level_io_level(val)
+    }
+
+    #[inline]
+    pub fn io_loop_back(&self) -> u32 {
+        self.0.io_loop_back()
+    }
+
+    #[inline]
+    pub fn set_io_loop_back(&mut self, val: u32) {
+        self.0.set_io_loop_back(val)
+    }
+}
+
+/// @brief PCNT channel
+#[derive(Debug)]
+pub struct PcntChannel(pcnt_channel_handle_t);
+
+impl PcntChannel {
+    /// @brief Set channel actions when edge signal changes (e.g. falling or rising edge occurred).
+    ///        The edge signal is input from the `edge_pin` configured in `PcntChanConfig`.
+    ///        We use these actions to control when and how to change the counter value.
+    ///
+    /// @param[in] pos_act Action on posedge signal
+    /// @param[in] neg_act Action on negedge signal
+    /// @return
+    ///      - (): on success
+    ///      - EspError: on failure
+    pub fn set_edge_action(&self,pos_act: PcntEdgeAction, neg_act: PcntEdgeAction) -> Result<(), EspError> {
+        unsafe {
+            esp!(pcnt_channel_set_edge_action(self.0, pos_act.into(), neg_act.into()))
+        }
+    }
+
+    /// @brief Set channel actions when level signal changes (e.g. signal level goes from high to low).
+    ///        The level signal is input from the `level_gpio_num` configured in `pcnt_chan_config_t`.
+    ///        We use these actions to control when and how to change the counting mode.
+    ///
+    /// @param[in] chan PCNT channel handle created by `pcnt_new_channel()`
+    /// @param[in] high_act Action on high level signal
+    /// @param[in] low_act Action on low level signal
+    /// @return
+    ///      - (): on success
+    ///      - EspError: on failure
+    pub fn set_level_action(&self, high_act: PcntLevelAction, low_act: PcntLevelAction) -> Result<(), EspError> {
+        unsafe {
+            esp!(pcnt_channel_set_level_action(self.0, high_act.into(), low_act.into()))
+        }
+    }
+}
+
+impl Drop for PcntChannel {
+    fn drop(&mut self) {
+        unsafe {
+            esp!(pcnt_del_channel(self.0)).unwrap();
+        }
+    }
+}
+
+#[derive(Debug, Default, Copy, Clone)]
+pub struct PcntUnitFlags(pcnt_unit_config_t__bindgen_ty_1);
```

Comment:
Ditto: depending on the size of the struct, maybe we don't want `Copy` and we probably want a `const fn new() -> Self`

Reply:
it's only 32 bits